pkg/agent/controller/traceflow/traceflow_controller_test.go

```diff
@@ -568,8 +848,89 @@ func TestProcessTraceflowItem(t *testing.T) {
 	tfc.mockOFClient.EXPECT().InstallTraceflowFlows(tc.tf.Status.DataplaneTag, tc.tf.Spec.LiveTraffic, tc.tf.Spec.DroppedOnly, tc.receiverOnly, nil, tc.ofPort, crdv1alpha1.DefaultTraceflowTimeout)
 	tfc.mockOFClient.EXPECT().SendTraceflowPacket(tc.tf.Status.DataplaneTag, tc.packet, tc.ofPort, int32(-1))
 
-	tfc.enqueueTraceflow(tc.tf)
-
 	got := tfc.processTraceflowItem()
 	assert.Equal(t, tc.expected, got)
 }
+
+func TestValidateTraceflow(t *testing.T) {
+	tcs := []struct {
+		name               string
+		tf                 *crdv1alpha1.Traceflow
+		antreaProxyEnabled bool
+		expectedErr        string
+	}{
+		{
+			name: "AntreaProxy feature disabled with destination as service",
+			tf: &crdv1alpha1.Traceflow{
+				Spec: crdv1alpha1.TraceflowSpec{
+					Destination: crdv1alpha1.Destination{
+						Service: "svcTest",
+					},
+				},
+			},
+			expectedErr: "using Service destination requires AntreaProxy feature enabled",
+		},
+		{
+			name: "invalid destination IPv4",
+			tf: &crdv1alpha1.Traceflow{
+				Spec: crdv1alpha1.TraceflowSpec{
+					Destination: crdv1alpha1.Destination{
+						IP: "192.168.1.300",
+					},
+				},
+			},
+			antreaProxyEnabled: true,
+			expectedErr:        "destination IP is not valid: 192.168.1.300",
+		},
+		{
+			name: "AntreaProxy feature disabled with ClusterIP destination",
+			tf: &crdv1alpha1.Traceflow{
+				Spec: crdv1alpha1.TraceflowSpec{
+					Destination: crdv1alpha1.Destination{
+						IP: "10.96.1.1",
+					},
+				},
+			},
+			expectedErr: "using ClusterIP destination requires AntreaProxy feature enabled",
+		},
+	}
+
+	for _, tt := range tcs {
+		t.Run(tt.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, features.DefaultFeatureGate, features.AntreaProxy, tt.antreaProxyEnabled)()
+
+			tfc := newFakeTraceflowController(t, nil, []runtime.Object{tt.tf}, nil, nil, nil, nil)
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			tfc.crdInformerFactory.Start(stopCh)
+			tfc.crdInformerFactory.WaitForCacheSync(stopCh)
+
+			err := tfc.validateTraceflow(tt.tf)
+			assert.ErrorContains(t, err, tt.expectedErr)
+		})
+	}
+}
+
+func TestDeleteTraceflow(t *testing.T) {
+	tf := &crdv1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{Name: "tf", UID: "uid"},
+		Spec: crdv1alpha1.TraceflowSpec{
+			Source: crdv1alpha1.Source{
+				Namespace: pod1.Namespace,
+				Pod:       pod1.Name,
+			},
+			Destination: crdv1alpha1.Destination{
+				IP: "192.168.1.200",
+			},
+		},
+	}
+
+	tfc := newFakeTraceflowController(t, []runtime.Object{&pod1}, []runtime.Object{tf}, nil, nil, nil, nil)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	tfc.crdInformerFactory.Start(stopCh)
+	tfc.crdInformerFactory.WaitForCacheSync(stopCh)
+
+	err := tfc.traceflowClient.CrdV1alpha1().Traceflows().Delete(context.TODO(), tf.Name, *metav1.NewDeleteOptions(0))
```

Comment:
I do not understand what you are testing here. You are deleting a `Traceflow` resource from the API (or rather from the fake client), but I don't think this is exercising any Antrea code.

Reply:
It is to cover lines of this method https://github.com/antrea-io/antrea/blob/6590c2adeef85c6b234ec5775caa7a612b16250f/pkg/agent/controller/traceflow/traceflow_controller.go#L202

Reply:
this is not a good test, it doesn't do anything. We should not add tests to artificially increase coverage measurements, if these tests do not verify anything.

Reply:
removed.